tt_metal/impl/buffers/buffer.cpp

```diff
@@ -157,7 +157,7 @@ Buffer::Buffer(Device *device, uint64_t size, uint64_t page_size, const BufferTy
     validate_buffer_size_and_page_size(size, page_size, buffer_type, buffer_layout, shard_parameters);
     if(is_sharded(buffer_layout)){
```

Comment:
Can't remember if we talked about this, but are you planning to create a separate class for Sharded buffers?

Reply:
ya will do in another PR. I'll file an issue